lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -158,8 +160,14 @@ def add_creditcard(post, payment, options)
         pm_fields[:number] = payment.number
         pm_fields[:expiration_month] = payment.month.to_s
         pm_fields[:expiration_year] = payment.year.to_s
-        pm_fields[:cvv] = payment.verification_value.to_s
         pm_fields[:name] = "#{payment.first_name} #{payment.last_name}"
+
+        if stored_credential
+          pm_fields[:recurrence_type] = stored_credential[:reason_type] if stored_credential[:reason_type]
+          pm_fields[:network_reference_id] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
```

Comment:
We should be populating `network_reference_id` with `original_network_transaction_id` instead of `network_transaction_id` since we update the latter on each subsequent transaction.

> Note: For all subsequent payments in the subscription, you must use the network reference ID from the first payment. The response in each subsequent payment contains a different network reference ID, which is not for use in requests.

Reply:
change is updated 👍